streamer/src/sendmmsg.rs

```diff
@@ -166,22 +183,28 @@ where
     S: Borrow<SocketAddr>,
     T: AsRef<[u8]>,
 {
-    let size = packets.len();
-    let mut iovs = vec![MaybeUninit::uninit(); size];
-    let mut addrs = vec![MaybeUninit::zeroed(); size];
-    let mut hdrs = vec![MaybeUninit::uninit(); size];
+    const MAX_IOV: usize = libc::UIO_MAXIOV as usize;
+
+    if packets.len() > MAX_IOV {
+        return Err(SendPktsError::IoError(
+            io::Error::new(io::ErrorKind::InvalidInput, "batch size exceeds UIO_MAXIOV"),
+            packets.len(),
+        ));
+    }
```

Comment:
> The value specified in vlen is capped to UIO_MAXIOV (1024).

This line is from [here](https://man7.org/linux/man-pages/man2/sendmmsg.2.html). Do you know if we would have errored previously if someone called `batch_send()` with `packets.len() > 1024` ? We use `batch_send()` in a handful of places; I didn't look exhaustively but the one or two I looked at look to log the error. Just wanting to make sure we aren't failing in a new way.

Reply:
Regardless of the answer to the question above, I think we could just wrap the entire function like this and just loop internally so all of the various callers don't have to worry about doing so:
```rust
const MAX_IOV: usize = libc::UIO_MAXIOV as usize;
for packet_chunk in packets.chunks(MAX_IOV) {
    let mut iovs = [MaybeUninit::uninit(); MAX_IOV];
    ...
```

Reply:
I'm not positive, but I think `fn sendmmsg_retry` would generally succeed when receiving greater than libc::UIO_MAXIOV packets.

Looks like it would try to send all the packets, and if >1024, would send the first 1024, hit an "error", jump to packet 1024, try to send all the packets again, etc. Only if we received a -1 response would we consider it an actual error to be returned. Even then, it skips over that packet and keeps trying to send the rest